holoviews/core/data/__init__.py

```diff
@@ -68,25 +68,37 @@ class DataConversion(object):
     def __init__(self, element):
         self._element = element
 
-    def __call__(self, new_type, kdims=None, vdims=None, mdims=None,
+    def __call__(self, new_type, kdims=None, vdims=None, groupby=None,
                  sort=False, **kwargs):
         """
-        Generic conversion method for Column based types. Supply the
-        Columns based type to convert to and optionally the
-        key dimensions (kdims), value dimensions (vdims) and HoloMap
-        key dimensions (mdims). Converted Columns can be automatically
-        sorted via the sort option and kwargs can be passed through.
+        Generic conversion method for Dataset based Element
+        types. Supply the Dataset Element type to convert to and
+        optionally the key dimensions (kdims), value dimensions
+        (vdims) and the dimensions.  to group over. Converted Columns
+        can be automatically sorted via the sort option and kwargs can
+        bepassed through.
         """
+        if 'mdims' in kwargs:
+            if groupby:
+                raise ValueError('Cannot supply both mdims and groupby')
+            else:
+                self._element.warning("mdims keyword has been renamed to "
+                                      "groupby and will be deprecated in "
+                                      "version 1.8.")
```

Comment:
"will be deprecated" is ambiguous; isn't it now deprecated already?  Plus it seems like we shouldn't promise a version 1.8 we might not have (in favor of 2.0). How about:

``mdims keyword has been renamed to groupby; the name mdims is deprecated and will be removed after version 1.7.``